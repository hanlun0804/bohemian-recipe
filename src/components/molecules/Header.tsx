import React from 'react'
import Logo from '../../assets/BR-logo-tr.png'

const Header: React.FC = () => {
    return (
        <div className="w-full bg-black flex items-center justify-center py-2 md:py-2">
            <div className="h-16 md:h-28">
                <a href="http://www.vg.no" className="header-logo-link">
                    <img className="h-full" src={Logo} alt="Logo" />
                </a>
            </div>
        </div>
    )
}

export default Header
