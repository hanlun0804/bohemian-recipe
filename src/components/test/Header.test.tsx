import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import { toMatchImageSnapshot } from 'jest-image-snapshot'
import Header from '../molecules/Header'

declare module 'vitest' {
    interface Assertion<T> {
        toMatchImageSnapshot(): T
    }
}

expect.extend({ toMatchImageSnapshot })

describe('Header', () => {
    it('Renders header with logo', () => {
        render(<Header />)
        const logoImage = screen.getByAltText(/Logo/i)
        expect(logoImage).toBeInTheDocument()

        expect(document.body).toMatchSnapshot()
    })
})
