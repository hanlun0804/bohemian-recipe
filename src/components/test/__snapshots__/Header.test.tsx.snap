// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Renders header with logo 1`] = `
<body>
  <div>
    <div
      class="w-full bg-black flex items-center justify-center py-2 md:py-2"
      data-testid="test-header"
    >
      <div
        class="h-16 md:h-28"
      >
        <a
          class="header-logo-link"
          href=""
        >
          <img
            alt="Logo"
            class="h-full"
            data-testid="cypress-logo"
            src="/project2/src/assets/BR-logo-tr.png"
          />
        </a>
      </div>
    </div>
  </div>
</body>
`;
